pkgs/development/libraries/cfitsio/default.nix

```
@@ -1,19 +1,32 @@
-{ fetchurl, stdenv }:
+{ fetchurl, stdenv
 
- stdenv.mkDerivation {
-  name = "cfitsio-3.430";
+# Optional dependencies
+, bzip2 ? null }:
+
+stdenv.mkDerivation rec {
+  name = "cfitsio-${version}";
+  version = "3.450";
 
   src = fetchurl {
-    url = "ftp://heasarc.gsfc.nasa.gov/software/fitsio/c/cfitsio3430.tar.gz";
-    sha256 = "07fghxh5fl8nqk3q0dh8rvc83npnm0hisxzcj16a6r7gj5pmp40l";
+    url = "ftp://heasarc.gsfc.nasa.gov/software/fitsio/c/cfitsio${builtins.replaceStrings ["."] [""] version}.tar.gz";
+    sha256 = "0bmrkw6w65zb0k3mszaaqy1f4zjm2hl7njww74nb5v38wvdi4q5z";
   };
 
+  buildInputs = [ bzip2 ];
+
   patches = [ ./darwin-curl-config.patch ./darwin-rpath-universal.patch ];
 
+  configureFlags = stdenv.lib.optional (bzip2 != null) "--with-bzip2=${bzip2.dev}";
```

```suggestion
  configureFlags = stdenv.lib.optional (bzip2 != null) "--with-bzip2=${bzip2.out}";
```

Then you wouldn't need the postFixup fix and libcfitsio is still linked to the correct libbz2
```
$ ldd result/lib/libcfitsio.so
[...]
        libbz2.so.1 => /nix/store/wlivxbc4ny21gzq9a561bwx9b4p08m4m-bzip2-1.0.6.0.1/lib/libbz2.so.1 (0x00007f331f773000)
```

---

Much better, thanks! I didn't realize the bzip2 headers could be resolved via `NIX_CFLAGS_COMPILE`, which was why I used `bzip2.dev`.